src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(health, name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  removeDead() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(health, name);
+  }
+
+  bite(individual) {
+    if (individual instanceof Herbivore && !individual.hidden) {
+      individual.health -= 50;
+
+      if (individual.health <= 0) {
+        individual.removeDead();
+      }
+    }
```

Comment:
[CODE PERFORMANCE] - To adhere to the checklist, avoid calling `removeDead` method within the `bite` method. Instead, you can filter the `Animal.alive` array directly within the `bite` method after the health deduction to remove dead animals, which is more efficient and avoids the issue of having to call a method on an individual animal. Use the `filter` method to remove any animals with health less than or equal to `0`.